src/data_c.c4

```diff
@@ -464,6 +529,32 @@ SHMEM_PROF_DEF_CTX_PUT_N_SIGNAL_NBI(`mem')
     }                                                        \
   }
 
+#define SHMEM_DEF_IBPUT_N(NAME,SIZE)                         \
+  void SHMEM_FUNCTION_ATTRIBUTES                             \
+  SHMEMX_FUNC_PROTOTYPE(ibput##NAME, void *target,           \
+                       const void *source, ptrdiff_t tst,    \
+                       ptrdiff_t sst, size_t bsize, size_t nblocks, int pe) \
+    long completion = 0;                                     \
+    SHMEM_ERR_CHECK_INITIALIZED();                           \
+    SHMEM_ERR_CHECK_PE(pe);                                  \
+    SHMEM_ERR_CHECK_CTX(ctx);                                \
+    SHMEM_ERR_CHECK_SYMMETRIC(target,                        \
+                           (SIZE) * ((nblocks-1) * tst + bsize)); \
+    SHMEM_ERR_CHECK_NULL(source, nblocks);                   \
+    SHMEM_ERR_CHECK_STRIDE_GTE_BSIZE(tst, bsize);            \
+    SHMEM_ERR_CHECK_STRIDE_GTE_BSIZE(sst, bsize);            \
+    SHMEM_ERR_CHECK_OVERLAP(target, source,                  \
+                        (SIZE) * ((nblocks-1) * tst + bsize), \
+                        (SIZE) * ((nblocks-1) * sst + bsize), 0); \
+    for ( ; nblocks > 0 ; --nblocks) {                       \
+      shmem_internal_put_nb(ctx, target, source,             \
+                            bsize * (SIZE), pe,              \
+                            &completion);                    \
+      target = (uint8_t*)target + tst*(SIZE);                \
+      source = (uint8_t*)source + sst*(SIZE);                \
```

Comment:
Just nitpick: could you use similar spacing for `*` as in other lines?

Reply:
Sure thing! I had tried to keep the styling consistent with the existing implementations of iput/iget. Should I make the spacing for `*` consistent in those functions as well?

Reply:
Oh, I see. Yeah, that would make sense. Thanks.